Patch-Source: https://github.com/amphp/ext-uv/pull/124
From 7edab710f5295c0ca5e04e2618fde8678d163ca3 Mon Sep 17 00:00:00 2001
From: Andy Postnikov <apostnikov@gmail.com>
Date: Mon, 26 Oct 2026 15:21:22 +0100
Subject: [PATCH] Fix -Wformat compiler warnings in zif_uv_timer_start

Use ZEND_LONG_FMT instead of %lld for zend_long variables (timeout and repeat)
to ensure proper 32-bit and 64-bit cross-platform compatibility.
---
 php_uv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/php_uv.c b/php_uv.c
index fdd285b..97f7158 100644
--- a/php_uv.c
+++ b/php_uv.c
@@ -4127,12 +4127,12 @@ PHP_FUNCTION(uv_timer_start)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (timeout < 0) {
-		php_error_docref(NULL, E_WARNING, "timeout value have to be larger than 0. given %lld", timeout);
+		php_error_docref(NULL, E_WARNING, "timeout value have to be larger than 0. given " ZEND_LONG_FMT, timeout);
 		RETURN_FALSE;
 	}
 
 	if (repeat < 0) {
-		php_error_docref(NULL, E_WARNING, "repeat value have to be larger than 0. given %lld", repeat);
+		php_error_docref(NULL, E_WARNING, "repeat value have to be larger than 0. given " ZEND_LONG_FMT, repeat);
 		RETURN_FALSE;
 	}
 
